### PR TITLE
Handle page up/down events on the table

### PR DESCRIPTION
**Pull request by shysteph**
I would expect if I clicked on the table and hit page up or page down the table would scroll.  Unfortunately it only seems to work if I carefully get focus on the scrollbar itself.

## Expected Behavior
I hit Page Down and the table scrolls down one page
## Current Behavior
I hit Page Down and nothing happens

## Possible Solution
PR incoming.  Add a _onKeyDown handler to FixedDataTable and call its _onScroll callback with a page worth of pixels.  

## Steps to Reproduce (for bugs)
http://schrodinger.github.io/fixed-data-table-2/example-object-data.html
Click on the table and page up/down.  Nothing happens.  Click very carefully on the scrollbar and repeat.  Page goes up and down.
Tested in chrome unstable + firefox on linux.

## Context
I want to use a table?  A big table.  I want to be able to navigate said big table.

## Your Environment
* Version used: a80065fd1fbc9e0bb0822a8e60dd07b9e2321384  - and - 0.8.0 (also shows up in fixed-data-table@^0.6.4)
* Browser Name and version: Version 61.0.3163.25 (Official Build) dev (64-bit)
* Operating System and version (desktop or mobile): Fedora 25
* Link to your project: Internal tool

### CODE REVIEW

File: src/FixedDataTable.js [hunk 1]
```diff
@@ -679,6 +690,8 @@ var FixedDataTable = createReactClass({
           cx('fixedDataTableLayout/main'),
           cx('public/fixedDataTable/main'),
         )}
+        tabIndex={0}
```
Comment (wcjordan) [inline src/FixedDataTable.js]: Was it not possible to trigger events keyboard events on the grid w/o it?

File: src/FixedDataTable.js [hunk 2]
```diff
@@ -456,6 +456,16 @@ var FixedDataTable = createReactClass({
     );
   },
 
+  _onKeyDown(event) {
+    if(event.key === "PageDown") {
```
Comment (wcjordan) [inline src/FixedDataTable.js]: nit single quotes here & around PageUp.
Also space after 0, in both _onScroll calls

File: src/FixedDataTable.js (hunk 1 above)
Comment (shysteph) [inline src/FixedDataTable.js]: It needed that to allow the component to get keyboard focus.  Scrollbar.js does the same thing.

**Comment by wcjordan** [comment]
Nice looks good, I'm going to try this out Monday and we should be able to merge in early next week.  Thanks for the contribution!

**Comment by wcjordan** [comment]
Looks good.  Could you flag this behind a prop and default it to off (just in case anyone doesn't want the change)?  Then merge in master and I'll merge and do a patch release.

I'll switch the flag to default to true on the next minor release.

**Comment by shysteph** [comment]
I already started writing that assuming I would need another pr.  I have a few places that need everything working and a few where I need page up/down.  Entirely off is a logical continuation.


I'll likely have it done tomorrow morning.


On Aug 24, 2017 5:42 PM, Chris Jordan <notifications@github.com> wrote:

Looks good. Could you flag this behind a prop and default it to off (just in case anyone doesn't want the change)? Then merge in master and I'll merge and do a patch release.

I'll switch the flag to default to true on the next minor release.

File: src/FixedDataTable.js [hunk 3]
```diff
@@ -556,6 +594,7 @@ var FixedDataTable = createReactClass({
 
       scrollbarYHeight = Math.max(0, footOffsetTop - bodyOffsetTop);
     }
+    this._scrollbarYHeight = scrollbarYHeight;
```
Comment (KamranAsif) [inline src/FixedDataTable.js]: Is this related to this change?

Comment (shysteph) [reply src/FixedDataTable.js]: Sorry, a few other projects came up.  I agree that it needs a flag before being merged.  I plan to add it as soon as I have a moment.  The saving of the computed scrollbarYHeight is part of the change.  It is used in the event handler to figure out how tall "1 page" is for page up/page down.

Comment (KamranAsif) [reply src/FixedDataTable.js]: Can we save this to state instead of having it as a property?

Comment (shysteph) [reply src/FixedDataTable.js]: The code that computes that value is in render.  If memory serves react really doesn't like it when you update state inside of render.  Also it isn't actually part of the UI state, more a computed value that has been cached.  It never gets used to render anything which is why I was ok with using a property.

**Comment by shysteph** [comment]
I added the flags as requested.  I do not have a good way to move the scrollbar height into the state and also I do not think it belongs there.

**Comment by wcjordan** [comment]
Thanks for the updates!  I'll cleanup the scrollbar height variable on the v1.0 branch.
